src/test/java/pages/HomePage.java

```diff
@@ -0,0 +1,41 @@
+package pages;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+
+
+
+
+public class HomePage extends BasePage {
+
+    public HomePage (WebDriver givenDriver) {
+        super(givenDriver);
+    }
+
+
+    @FindBy (css = ".playlist:nth-child(3)")
+    WebElement firstPlaylist;
+    @FindBy (css = "[name='name']");
+    WebElement playlistNameField;
+    @FindBy (css = "div.success.show");
+    WebElement renamePlaylistSuccessMsg;
```

Comment:
You should use encapsulation of non-static and non-final fields.

If a variable is used only by other methods of that class, it should be declared as private.

I recommend:  make all the locators and WebElements private, because private keyword in java allows most restrictive access to variables and methods and offer strongest form of Encapsulation. 

For example

    ```
    private By shuffleBtnLocator = By.cssSelector(".btn-shuffle-all");
    private By firstSongLocator = By.cssSelector(".play");

```
   @FindBy(css = "img.avatar")
    private WebElement avatar
```

![POM_vs_Factory_1](https://github.com/Test-Pro-Bootcamp/A49-QA-Automation-Framework/assets/104761482/31da6c5c-6534-4489-9d75-a5e737b23bc5)

![POM_vs_Factory_2](https://github.com/Test-Pro-Bootcamp/A49-QA-Automation-Framework/assets/104761482/2d9c7371-5173-4b0c-ad59-05f08f68a09b)